shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -911,8 +952,22 @@ - (void)dispatchTouches:(NSSet*)touches
     return;
   }
 
+  // When UIApplicationSupportsIndirectInputEvents is set in Info.plist, mouse clicks will be
```

Comment:
Do we ever unset `UIApplicationSupportsIndirectInputEvents`? If not we might change it to
> With UIApplicationSupportsIndirectInputEvents set in Info.plist

Reply:
I wrote it that way since `UIApplicationSupportsIndirectInputEvents` is false by default, I need to make a PR to migrate it to true. But whether that key is true or false, each mouse click will always have a unique identifier, since it comes in as a `UITouch`. An accommodation is needed since mouse events with device kind touch don't trigger MouseRegion / mouse tracker and there are no side effects from creating a lot of single-use devices in the framework.

Reply:
Ok, firstly, what exactly is the "mouse device" you're talking about?

Secondly, do you mean that the behavior as observed in Flutter will vary based on `UIApplicationSupportsIndirectInputEvents`, so that if it's false the resulting events will have type touch, and if it's true the events will have type mouse?

Reply:
When I say mouse device, I guess I mean a sequence of pointer events with a unique identifier and device-kind `kMouse`. Those sequences have different side effects in the framework than sequences with a unique identifier but device-kind `kTouch`. 

For point 2, yes, that's correct, the event type will change depending on `UIApplicationSupportsIndirectInputEvents`, as well as a [few other things](https://developer.apple.com/documentation/bundleresources/information_property_list/uiapplicationsupportsindirectinputevents?language=objc):

Reply:
I see. I've also found https://developer.apple.com/documentation/uikit/pointer_interactions, which explains that an indirect pointer stands for the trackpad, while direct pointer stands for finger touches or pencils. We probably want to document this behavioral difference in the framework. 

Then for the documentation, how about this:

> If the `UIApplicationSupportsIndirectInputEvents` in `Info.plist` returns YES, then the platform dispatches indirect pointer touches (trackpad clicks) as `UITouch` with a type of `UITouchTypeIndirectPointer` and a different identifier for each click. They are translated into Flutter pointer events with `type` of `kMouse` and different device IDs. These devices must be terminated with `kRemove` events then the touches end, otherwise they will keep triggering hover events. 
>
> If the `UIApplicationSupportsIndirectInputEvents` in `Info.plist` returns NO, then the platform dispatches indirect pointer touches (trackpad clicks) as `UITouch` with a type of `UITouchTypeIndirectPointer` and a different identifier for each click. They are translated into Flutter pointer events with `type` of `kTouch` and different device IDs. Removing these devices is neither necessary nor harmful.
>
> Therefore Flutter always remove these devices. The `touches_to_remove_count` tracks how many remove events are needed in this group of touches to properly allocate space for the packet. The remove event of a touch is synthesized immediately after its normal event.

Reply:
Great summarization, I'll put it in.